src/TSHelper.ts

```diff
@@ -208,9 +208,19 @@ export class TSHelper {
     public static isInTupleReturnFunction(node: ts.Node, checker: ts.TypeChecker): boolean {
         const declaration = TSHelper.findFirstNodeAbove(node, ts.isFunctionLike);
         if (declaration) {
-            let functionType: ts.Type;
+            let functionType: ts.Type | undefined;
```

Comment:
I don't quite understand how this does not break on line 299 if `functionType` is undefined. Maybe this should be initialized as `checker.getTypeAtLocation(declaration)` be default and then overwritten if appropriate?

Reply:
TS can infer that it can't be `undefined` at that point because every branching path sets it to a value. We could initialize it, but it adds unnecessary overhead for some code paths.

Reply:
Aha I see, well maybe just removing the `else` and moving the `if (functionType === undefined) { functionType = ` as a separate step after this if if statement would be clearner then.